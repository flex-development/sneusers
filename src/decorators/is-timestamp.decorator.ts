/**
 * @file Decorators - IsTimestamp
 * @module sneusers/decorators/IsTimestamp
 */

import { ValidateBy, type ValidationOptions } from 'class-validator'
import validator from './is-timestamp.constraint'

/**
 * Ensures a value is a [unix timestamp][1].
 *
 * [1]: https://unixtimestamp.com
 *
 * @see {@linkcode ValidationOptions}
 *
 * @decorator
 *
 * @param {ValidationOptions} [options={}] - Validation options
 * @return {PropertyDecorator} Property decorator
 */
const IsTimestamp = (options: ValidationOptions = {}): PropertyDecorator => {
  return ValidateBy(
    Object.assign({ constraints: [options], validator }, validator.options),
    options
  )
}

export default IsTimestamp
