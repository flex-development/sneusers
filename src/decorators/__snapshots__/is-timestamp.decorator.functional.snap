// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`functional:decorators/IsTimestamp > should cause validation failure if validation fails 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isTimestamp": "updated_at must be a valid unix timestamp",
    },
    "property": "updated_at",
    "target": TestModel {
      "updated_at": null,
    },
    "value": null,
  },
]
`;
